modules/fx/arrow-fx/src/test/kotlin/arrow/fx/QueueTest.kt

```diff
@@ -185,22 +184,11 @@ class QueueTest : UnitSpec() {
         }.unsafeRunSync()
       }
 
-      "$label - offering to a 0 capacity queue in deficit honours blocking strategy" {
-        IO.fx {
-          val q = !queue(0)
-          // flip from initial Surplus state to Deficit
-          val first = !q.take().fork(ctx)
-          // then clear previous taker while staying in Deficit
-          !q.offer(1)
-          !first.join()
-          val start = !effect { System.currentTimeMillis() }
-          val wontComplete = q.offer(2)
-          val received = !wontComplete.map { Some(it) }
-            .waitFor(100.milliseconds, default = just(None))
-          val elapsed = !effect { System.currentTimeMillis() - start }
-          !effect { received shouldBe None }
-          !effect { (elapsed >= 100) shouldBe true }
-        }.unsafeRunSync()
+      "$label - capacity must be a positive integer" {
+        queue(0).attempt().suspended().fold(
```

Comment:
Can we used a refined tipe for positive integers?

Reply:
Not without having a proper refinement system? Or did you have something in mind?

Reply:
Some Q&D wrapper without getting too much into refined types.

```
class Size private (val s: Int) {
 co {
  of invoke(s: Int): Option<Size>
 }
}
```

Reply:
Wouldn't take be as much of a hassle as what the current API have without compile time literal verification?

We could provide alternative factory methods which return `IO<Option<Queue<ForIO, A>>>`?

I'm opening another PR asap which adds the `peek` and `try` methods to `Queue` and removes the other implementation. I will also add a `QueueFactory` like we have for `MVar` and `Ref`. We can continue this discussion there.